src/main/xerus/monstercat/api/APIConnection.kt

```diff
@@ -73,24 +74,34 @@ class APIConnection(vararg path: String) : HTTPQuery<APIConnection>() {
 	/** @return null when the connection fails, else the parsed result */
 	fun getTracks() =
 		parseJSON(TrackResponse::class.java)?.results
-	
+
+	private var httpRequest: HttpUriRequest? = null
 	/** Aborts this connection and thus terminates the InputStream if active */
 	fun abort() {
-		httpGet?.abort()
+		httpRequest?.abort()
 	}
 	
 	// Direct Requesting
-	
-	private var httpGet: HttpGet? = null
-	fun execute() {
-		httpGet = HttpGet(uri)
-		response = execute(httpGet!!)
+
+	fun get(context: HttpClientContext? = null) {
+		httpRequest = HttpGet(uri)
+		response = execute(httpRequest!!, context)
+	}
+
+	fun post(request : HttpPost, context: HttpClientContext? = null) {
```

Comment:
Why do we even have different methods for this? I think there could be one which simply takes `request: HttpUriRequest`, just called `execute`, and it could be used by the `get` method as well

Reply:
05fd56b27c85faf3947adf010501a57e1564cf1a

Reply:
One execute() for everything. Just add your own HttpGet / HttpPost / HttpPut as argument to choose which request type to use.

Reply:
I renamed the old execute in the Companion because it had the same arguments as the new function.